packages/host/app/components/matrix/register-user.gts

```diff
@@ -713,12 +714,18 @@ export default class RegisterUser extends Component<Signature> {
 
       throw e;
     }
+
     // If access_token and device_id are present, RegisterResponse matches LoginResponse
     // except for the optional well_known field
-    if (auth.access_token && auth.device_id) {
+    if (
+      auth.access_token &&
+      auth.device_id &&
+      this.state.type === 'waitForEmailValidation'
```

Comment:
i'm curious why we needed this condition, there is no guaranteed order when email validation occurs and when the user provides the registration token--they can happen in either order. wouldn't this force a particular order?

Reply:
So given that in the app, the (optional) registration token input phase always appears before the email validation I thought the order is guaranteed in a way where `waitForEmailValidation` is the last phase.

I actually added `this.state.type === 'waitForEmailValidation'` to satisfy the type check where `token` is currently only defined in `type: 'sendToken'` and `type: 'waitForEmailValidation';`. Otherwise the type checker will complain `token` doesn't exist in other types, for example when `type == 'register'`

So given `token` doesn't exist in all state types I think we have 2 options, either to narrow down the possible states in which `initializeNewUser(..., token)` can get called, or simply check if registration token is in state and use it if it is, something like:

```
if (auth.access_token && auth.device_id) {
      await this.matrixService.initializeNewUser(
        auth as LoginResponse,
        this.state.name,
        'token' in this.state ? this.state.token : undefined,
      );
    }
```

Reply:
I added a comment on why I added this condition in https://github.com/cardstack/boxel/pull/1897/commits/7f0bea5e4dcb4a3576b4e3b53a5a6ec648040234. If this is still concerning we can talk more about it in a pairing session but for now I'll go ahead and merge this.